dracut/scripts/coreos-installer-growfs

```diff
@@ -4,39 +4,28 @@ set -euo pipefail
 path=/growroot
 udevadm settle
 
-mkdir -p ${path}
-mount -t xfs /dev/disk/by-label/root ${path}
-
-majmin=$(findmnt -nvr -o MAJ:MIN "$path")
-
-cryptsetup=0
-dm_name=""
-src=$(findmnt -nvr -o SOURCE "$path")
-if [[ "${src}" =~ /dev/mapper ]]; then
-    dm_name=$(dmsetup info ${src} -C -o name --noheadings) || true
-    if [[ "$(dmsetup info ${dm_name} -C -o uuid --noheadings)" =~ "CRYPT-LUKS2" ]]; then
-        majmin=$(dmsetup table ${dm_name} | cut -d " " -f7)
-        cryptsetup=1
-    fi
+disk=$(lsblk -ln -o NAME,TYPE | grep disk | awk '{print $1}')
+if [[ -z ${disk} ]]; then
+    exit 1
 fi
+growpart /dev/${disk} 4
 
-devpath=$(realpath "/sys/dev/block/$majmin")
-partition=$(cat "$devpath/partition")
-parent_path=$(dirname "$devpath")
-parent_device=/dev/$(basename "${parent_path}")
-
-growpart "${parent_device}" "${partition}" || true
+crypt=$(lsblk -ln -o NAME,TYPE | grep crypt | awk '{print $1}')
+if [[ -z ${crypt} ]]; then
+    exit 1
+fi
+cryptsetup resize ${crypt}
+lvm pvresize /dev/mapper/${crypt}
 
-if [ "${cryptsetup}" -eq 1 ]; then
-    id="$(cryptsetup luksDump "${parent_device}${partition}" | sed -rn 's|^\s+([0-9]+): clevis|\1|p')"
-    token=$(cryptsetup token export --token-id "${id}" "${parent_device}${partition}")
-    key=$(echo ${token} \
-        | jose fmt -j- -Og jwe -o- \
-        | jose jwe fmt -i- -c \
-        | tr -d '\n' \
-        | clevis decrypt)
+lvm vgchange -ay rootvg
 
-    echo $key | cryptsetup resize "${dm_name}" -
+if [[ ! -e /dev/disk/by-label/root ]]; then
+    dm=$(blkid --label root)
+    if [[ -z ${dm} ]]; then
+        exit 1
+    fi
+    ln -s ${dm} /dev/disk/by-label/root
 fi
 
-xfs_growfs $path
+mkdir -p ${path}
+mount -t xfs /dev/disk/by-label/root ${path}
```

Comment:
just noticed these 2 lines, they're not doing anything useful afaict, @7flying can you confirm? also in general, coreos-installer-dracut now only supports crypted device or else it'll fail. The previous code handled both encrypted and unencrypted images which we should strive to maintain I guess